"bind: address already in use" after switching to gRPC-Web

**Describe the bug**

After switching from an older version of ArgoCD to 0.12 we're encountering issues with the cli when trying to `sync` & `wait` successive applications on a releasing step. Our script looks like this:

```
#!/bin/bash

set -euo pipefail

# parameters
server=myserver.com

# download latest argocd cli.
curl -sSL -o /usr/local/bin/argocd https://${server}/download/argocd-linux-amd64
chmod +x /usr/local/bin/argocd

# tell argocd to sync apps.
apps=(${APPS:-})
for app in "${apps[@]}"; do
    echo "Applying changes in '${app}'..."
    argocd app sync ${app} --grpc-web --server ${server}
    argocd app wait ${app} --grpc-web --server ${server}
done
```

**To Reproduce**

Cannot say, as it fails randomly with this error:

```
10:34:20 Applying changes in 'edge-proxy'...
10:34:20 time="2019-04-12T10:34:20Z" level=fatal msg="Failed to establish connection to myserver.com:443: listen unix /tmp/argocd-1555065260.sock: bind: address already in use"
10:34:20 make: *** [release] Error 1
10:34:20 Makefile:33: recipe for target 'release' failed
```

**Expected behavior**

Finish successfully :)

Thanks for such a cool project btw!

--- a/pkg/apiclient/grpcproxy.go
+++ b/pkg/apiclient/grpcproxy.go
@@ -11,7 +11,6 @@ import (
 	"os"
 	"strconv"
 	"strings"
-	"time"
 
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
@@ -20,6 +19,7 @@ import (
 
 	argocderrors "github.com/argoproj/argo-cd/errors"
 	"github.com/argoproj/argo-cd/util"
+	"github.com/argoproj/argo-cd/util/rand"
 )
 
 const (
@@ -103,7 +103,7 @@ func (c *client) executeRequest(fullMethodName string, msg []byte, md metadata.M
 }
 
 func (c *client) startGRPCProxy() (*grpc.Server, net.Listener, error) {
-	serverAddr := fmt.Sprintf("%s/argocd-%d.sock", os.TempDir(), time.Now().Unix())
+	serverAddr := fmt.Sprintf("%s/argocd-%s.sock", os.TempDir(), rand.RandString(16))
 	ln, err := net.Listen("unix", serverAddr)
 
 	if err != nil {
@@ -191,10 +191,9 @@ func (c *client) useGRPCProxy() (net.Addr, io.Closer, error) {
 		c.proxyUsersCount = c.proxyUsersCount - 1
 		if c.proxyUsersCount == 0 {
 			c.proxyServer.Stop()
-			err := c.proxyListener.Close()
 			c.proxyListener = nil
 			c.proxyServer = nil
-			return err
+			return nil
 		}
 		return nil
 	}}, nil

